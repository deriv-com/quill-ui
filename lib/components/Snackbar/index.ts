export * from "./snackbar";
export * from "./snackbar-controller";
