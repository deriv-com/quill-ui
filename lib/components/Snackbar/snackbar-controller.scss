@import "@deriv-com/quill-tokens/dist/breakpoints.scss";

.snackbar {
    &--container {
        display: flex;
        justify-content: center;
    }
}
