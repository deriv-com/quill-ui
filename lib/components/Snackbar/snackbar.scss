@import "@deriv-com/quill-tokens/dist/breakpoints.scss";

.snackbar {
    display: flex;
    width: calc(100vw - var(--core-spacing-1600));
    max-width: 560px;
    min-height: var(--component-snackbar-height-min);
    padding-inline: var(--component-snackbar-spacing-padding-md)
        var(--component-snackbar-spacing-padding-sm);
    padding-block: var(--component-snackbar-spacing-padding-sm);
    align-items: center;
    gap: var(--component-snackbar-spacing-gap);
    border-radius: var(--component-snackbar-border-radius);
    border: var(--component-snackbar-border-width) solid
        var(--component-snackbar-border-color);
    background: var(--component-snackbar-bg);
    box-shadow: var(--core-elevation-shadow-330);
    position: fixed;
    z-index: 999;
    bottom: -16px;
    transition: bottom var(--motion-duration-snappy)
        var(--motion-easing-inandout);

    &--container {
        display: flex;
        justify-content: center;
    }

    &.slide-up {
        animation: slide-up var(--motion-duration-snappy)
            var(--motion-easing-inandout) forwards;
    }

    &.slide-down {
        animation: slide-down var(--motion-duration-snappy)
            var(--motion-easing-inandout) forwards;
    }

    @include breakpoint("sm") {
        display: inline-flex;
        width: var(--component-snackbar-width-max);
    }

    &__icon--container {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        flex-shrink: 0;
        & > svg {
            fill: var(--component-snackbar-icon);
        }
    }
    &__message {
        display: -webkit-box;
        -webkit-box-orient: vertical;
        -webkit-line-clamp: 2;
        flex: 1 0 0;

        text-overflow: ellipsis;
        overflow: hidden;
        word-wrap: break-word; /* Text gets broken up into another line for long words */
       
        @include breakpoint("sm") {
            -webkit-line-clamp: 1;
        }

        &--container {
            display: flex;
            padding-right: var(--semantic-spacing-general-sm);
            align-items: flex-start;
            flex: 1 0 0;
            min-width: var(
                --semantic-spacing-general-none
            ); /* Text broken up into another line for long words */
        }
    }

    & .button-label svg {
        display: flex;
    }
}
@keyframes slide-up {
    from {
        bottom: -100%;
    }
    to {
        bottom: var(--core-size-800);
    }
}

@keyframes slide-down {
    from {
        bottom: var(--core-size-800);
    }
    to {
        bottom: -100%;
    }
}
