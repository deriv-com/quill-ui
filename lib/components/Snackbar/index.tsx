import React, { ReactNode, useEffect, useState, HTMLAttributes } from "react";
import { Text } from "@components/Typography";
import "./snackbar.scss";
import clsx from "clsx";
import { Button } from "../Button";
import { LabelPairedXmarkSmBoldIcon } from "@deriv/quill-icons";

interface SnackbarProps extends HTMLAttributes<HTMLDivElement> {
    icon?: ReactNode;
    message: string;
    actionText?: string;
    hasCloseButton?: boolean;
    onClose: () => void;
    onActionClick?: () => void;
    isOpen: boolean;
}

export const Snackbar = ({
    icon: Icon,
    message,
    actionText,
    onActionClick,
    hasCloseButton = true,
    isOpen = false,
    onClose,
    ...rest
}: SnackbarProps) => {
    const animationSpeedObj = Object.freeze({
        fast: "fast",
        slow: "slow",
    });

    const [animationSpeed, setAnimationSpeed] = useState<
        keyof typeof animationSpeedObj
    >(animationSpeedObj.slow);
    useEffect(() => {
        if (isOpen) {
            setAnimationSpeed(animationSpeedObj.slow);
            const timer = setTimeout(() => {
                onClose?.();
            }, 3000);

            return () => {
                clearTimeout(timer);
            };
        }
    }, [isOpen]);

    const handleClose = () => {
        onClose?.();
        setAnimationSpeed(animationSpeedObj.fast);
    };

    const handleActionClick = () => {
        onActionClick?.();
        setAnimationSpeed(animationSpeedObj.fast);
    };
    return (
        <>
            {isOpen && (
                <div
                    className={clsx(
                        "snackbar",
                        animationSpeed === animationSpeedObj.fast
                            ? "fast-animation"
                            : "slow-animation",
                    )}
                    {...rest}
                >
                    {Icon && (
                        <div className="snackbar__icon--container">{Icon}</div>
                    )}
                    <div className="snackbar__message--container">
                        <Text className="snackbar__message" size="sm">
                            {message}
                        </Text>
                    </div>
                    {actionText && (
                        <Button
                            variant="tertiary"
                            label={actionText}
                            color="white"
                            onClick={handleActionClick}
                        />
                    )}
                    {hasCloseButton && (
                        <Button
                            variant="tertiary"
                            label={<LabelPairedXmarkSmBoldIcon />}
                            color="white"
                            onClick={handleClose}
                            data-testid="close-button"
                        />
                    )}
                </div>
            )}
        </>
    );
};
