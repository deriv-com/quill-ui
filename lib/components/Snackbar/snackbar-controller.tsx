import React from "react";
import { Snackbar, SnackbarProps } from "@components/Snackbar/snackbar";
import { useSnackbar } from "@hooks/useSnackbar";
import "./snackbar-controller.scss";

export const SnackbarController = () => {
    const { queue } = useSnackbar();
    return (
        <div className="snackbar--container">
            {queue.slice(0, 1).map((item: SnackbarProps) => {
                return (
                    <Snackbar
                        id={item.id}
                        isVisible={item.isVisible}
                        message={item.message}
                        actionText={item.actionText}
                        hasCloseButton={item.hasCloseButton}
                        key={item.id}
                        icon={item.icon}
                        onActionClick={item.onActionClick}
                    />
                );
            })}
        </div>
    );
};
