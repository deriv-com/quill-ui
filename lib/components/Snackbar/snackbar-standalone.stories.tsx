import React from "react";
import type { Meta, StoryObj } from "@storybook/react";
import { StandalonePlaceholderRegularIcon } from "@deriv/quill-icons/Standalone";
import { Snackbar } from "./snackbar";

const icons: Record<string, object | null> = {
    with_icon: <StandalonePlaceholderRegularIcon iconSize="sm" />,
    none: null,
};

const meta = {
    title: "Components/Snackbar/Standalone",
    component: Snackbar,
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
    tags: ["autodocs"],
    args: {
        icon: <StandalonePlaceholderRegularIcon iconSize="sm" />,
        isVisible: true,
        actionText: "",
        hasCloseButton: true,
    },
    parameters: {
        docs: {
            story: {
                height: "80px",
            },
        },
    },
    argTypes: {
        icon: {
            description:
                "Optional. There will be no icon if icon is not provided.",
            table: { type: { summary: "Reactnode | undefined" } },
            options: Object.keys(icons),
            mapping: icons,
            control: "radio",
        },
        id: { table: { disable: true } },
        isVisible: {
            description:
                "Required. This boolean is set and controlled by Provider. ",
        },
        message: {
            control: "text",
            description: "Required. Enter any message.",
        },
        actionText: {
            control: "text",
            description:
                "Optional. Action button will not be included if actionText is null or an empty string.",
            table: { type: { summary: "string | undefined" } },
        },
        onActionClick: {
            description:
                "Optional. Must be included if we have actionText to trigger action and close snackbar.",
        },
        onCloseAction: {
            description:
                "Optional. Only for standalone snackbar when you pass function for close button `onClick`.",
        },
        onSnackbarRemove: {
            description:
                "Optional. Callback for removing snackbar, will be triggered on close button click, on action text click, on timeout expiry and on dismissing the snackbar.",
        },
        hasCloseButton: {
            control: "boolean",
            description: "Optional. Set to true by default.",
        },
        hasFixedHeight: {
            control: "boolean",
            description: "Optional. Set to true by default.",
        },
    },
} satisfies Meta<typeof Snackbar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
    args: {
        message: "Message goes here",
    },
};

export const WithoutCloseButton: Story = {
    args: {
        message: "Message goes here",
        hasCloseButton: false,
    },
};

export const WithActionButton: Story = {
    args: {
        message: "Message goes here",
        actionText: "Action",
    },
};

export const WithRemoveCallback: Story = {
    args: {
        message: "Please, open the console",
        actionText: "Action",
        onSnackbarRemove: () => console.log("onSnackbarRemove was called"),
    },
};

export const WithoutFixedHeight: Story = {
    args: {
        icon: undefined,
        hasFixedHeight: false,
        message:
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut hendrerit, leo sed viverra consequat, quam nulla maximus est, et faucibus nunc urna sit amet ex. Donec sagittis fermentum finibus.",
    },
};
