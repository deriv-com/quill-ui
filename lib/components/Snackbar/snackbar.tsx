import React, { ReactNode, useEffect, HTMLAttributes, useRef } from "react";
import { Text } from "@components/Typography";
import clsx from "clsx";
import { Button } from "@components/Button";
import { LabelPairedXmarkSmBoldIcon } from "@deriv/quill-icons";
import { useSnackbar } from "@hooks/useSnackbar";
import { useOnClickOutside } from "usehooks-ts";
import { TDefaultColor } from "@types";
import "./snackbar.scss";

export interface SnackbarProps extends HTMLAttributes<HTMLDivElement> {
    icon?: ReactNode;
    id?: string;
    isVisible?: boolean;
    message: ReactNode;
    actionText?: string;
    hasCloseButton?: boolean;
    hasFixedHeight?: boolean;
    onActionClick?: () => void;
    onCloseAction?: () => void;
    onSnackbarRemove?: () => void;
    delay?: number;
    standalone?: boolean;
    status?: "fail" | "neutral";
}

export const Snackbar = ({
    icon: Icon,
    id,
    isVisible,
    message,
    actionText,
    onActionClick,
    onCloseAction,
    onSnackbarRemove,
    hasCloseButton = true,
    hasFixedHeight = true,
    delay,
    standalone = true,
    status = "neutral",
    ...rest
}: SnackbarProps) => {
    const { removeSnackbar } = useSnackbar();
    const timerRef = useRef<ReturnType<typeof setTimeout> | null>(null);
    const snackbarDelayedRemove = ({
        id,
        onSnackbarRemove,
        delay = 4000,
    }: {
        id: string;
        onSnackbarRemove?: () => void;
        delay?: number;
    }) => {
        return setTimeout(() => {
            removeSnackbar(id, onSnackbarRemove);
        }, delay);
    };

    const ref = useRef<HTMLDivElement>(null);

    useEffect(() => {
        if (id) {
            if (status === "neutral" || !hasCloseButton || delay !== undefined) {
                timerRef.current = snackbarDelayedRemove({
                    id,
                    onSnackbarRemove,
                    delay: delay || 4000,
                });
            }
        }

        return () => {
            if (timerRef.current) clearTimeout(timerRef.current);
        };
    }, [message, status, hasCloseButton, delay, id, onSnackbarRemove]);

    const handleClose = () => {
        onCloseAction?.();
        if (timerRef.current) clearTimeout(timerRef.current);
        id && removeSnackbar(id, onSnackbarRemove);
    };

    const handleActionClick = () => {
        onActionClick?.();
        handleClose();
    };

    id && useOnClickOutside(ref, () => removeSnackbar(id, onSnackbarRemove));

    if (standalone && !isVisible) return null;

    const color = {
        neutral: "white-black",
        fail: "white",
    };

    const buttonColor = color[status] as TDefaultColor;

    return (
        <div
            {...rest}
            className={clsx(
                "quill-snackbar",
                `quill-snackbar__status--${status}`,
                isVisible ? "slide-up" : "slide-up slide-down",
            )}
            ref={ref}
        >
            {Icon && (
                <div className={clsx("quill-snackbar__icon--container")}>
                    {React.isValidElement(Icon) &&
                        React.cloneElement(Icon, {
                            fill: `var(--component-snackbar-icon-${status})`,
                            ...Icon.props,
                        })}
                </div>
            )}
            <div className="quill-snackbar__message--container">
                <Text
                    className={clsx(
                        "quill-snackbar__message",
                        hasFixedHeight &&
                            "quill-snackbar__message--has-fix-height",
                    )}
                    size="sm"
                    style={{
                        color: `var(--component-snackbar-label-color-${status})`,
                    }}
                >
                    {message}
                </Text>
            </div>
            {actionText && (
                <Button
                    variant="tertiary"
                    label={actionText}
                    color={buttonColor}
                    onClick={handleActionClick}
                />
            )}
            {hasCloseButton && (
                <Button
                    variant="tertiary"
                    iconPosition="start"
                    icon={<LabelPairedXmarkSmBoldIcon />}
                    color={buttonColor}
                    size="md"
                    onClick={handleClose}
                    data-testid="close-button"
                />
            )}
        </div>
    );
};
