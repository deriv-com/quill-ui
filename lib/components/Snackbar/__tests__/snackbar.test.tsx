import { act, fireEvent, render, screen } from "@testing-library/react";
import { useSnackbar } from "../../../hooks/useSnackbar";
import { SnackbarProvider } from "../../../providers/snackbar/snackbarProvider";
import { SnackbarController } from "..";
import { SnackbarProps } from "../Snackbar";

const removeSnackbarMock = jest.fn();
jest.mock("../../../hooks/useSnackbar", () => ({
    useSnackbar: jest.fn(() => ({
        queue: [],
        addSnackbar: jest.fn(),
        removeSnackbar: jest.fn(),
    })),
}));

describe("Snackbar", () => {
    const testMessage = "test message";
    let defaultProps: Omit<SnackbarProps, "isShown">;
    beforeEach(() => {
        defaultProps = {
            icon: "",
            message: testMessage,
            actionText: "Action",
            hasCloseButton: true,
            id: "1",
        };
    });

    const renderComponent = () => {
        return render(
            <SnackbarProvider>
                <SnackbarController />
            </SnackbarProvider>,
        );
    };
    it("renders with default props", async () => {
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
        });
        const { container } = renderComponent();

        expect(screen.getByText(testMessage)).toBeInTheDocument();
        expect(screen.getByText("Action")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("renders correctly with custom icon", () => {
        defaultProps.icon = <img src="custom-icon.svg" alt="Custom Icon" />;
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
        });
        const { container } = renderComponent();
        expect(screen.getByAltText("Custom Icon")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("renders correctly without action button", () => {
        defaultProps.actionText = "";
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
        });
        const { container } = renderComponent();
        expect(screen.queryByText("Action")).not.toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("renders correctly without close button", () => {
        defaultProps.hasCloseButton = false;
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
        });
        const { container } = renderComponent();
        expect(screen.queryByTestId("close-button")).not.toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("calls removeSnackbar when close button is clicked", async () => {
        jest.useFakeTimers();
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
            removeSnackbar: removeSnackbarMock,
        });
        const { container } = renderComponent();

        act(() => {
            fireEvent.click(screen.getByTestId("close-button"));
            jest.advanceTimersByTime(100);
        });

        expect(removeSnackbarMock).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
        jest.useRealTimers();
    });
    it("calls removeSnackbar when action button is clicked", async () => {
        jest.useFakeTimers();
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
            removeSnackbar: removeSnackbarMock,
        });
        const { container } = renderComponent();

        act(() => {
            fireEvent.click(screen.getByText("Action"));
            jest.advanceTimersByTime(100);
        });

        expect(removeSnackbarMock).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
        jest.useRealTimers();
    });
    it("calls removeSnackbar after a certain duration when Snackbar is open", async () => {
        jest.useFakeTimers();
        (useSnackbar as jest.Mock).mockReturnValue({
            queue: [{ ...defaultProps }],
            removeSnackbar: removeSnackbarMock,
        });
        const { container } = renderComponent();
        jest.advanceTimersByTime(4000);

        expect(removeSnackbarMock).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
        jest.useRealTimers();
    });
});
