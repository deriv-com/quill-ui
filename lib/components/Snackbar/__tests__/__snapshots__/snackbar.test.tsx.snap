// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar calls removeSnackbar after a certain duration when Snackbar is open 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            Action
          </p>
        </span>
      </button>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
        data-testid="close-button"
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 11 22"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902"
                />
              </g>
              <defs>
                <clippath
                  id="bc20b7afc9245ecd8673f572bfeb5871__a"
                >
                  <path
                    d="M0 0h11v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar calls removeSnackbar when action button is clicked 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            Action
          </p>
        </span>
      </button>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
        data-testid="close-button"
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 11 22"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902"
                />
              </g>
              <defs>
                <clippath
                  id="bc20b7afc9245ecd8673f572bfeb5871__a"
                >
                  <path
                    d="M0 0h11v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar calls removeSnackbar when close button is clicked 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            Action
          </p>
        </span>
      </button>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
        data-testid="close-button"
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 11 22"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902"
                />
              </g>
              <defs>
                <clippath
                  id="bc20b7afc9245ecd8673f572bfeb5871__a"
                >
                  <path
                    d="M0 0h11v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar renders correctly with custom icon 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__icon--container"
      >
        <img
          alt="Custom Icon"
          src="custom-icon.svg"
        />
      </div>
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            Action
          </p>
        </span>
      </button>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
        data-testid="close-button"
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 11 22"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902"
                />
              </g>
              <defs>
                <clippath
                  id="bc20b7afc9245ecd8673f572bfeb5871__a"
                >
                  <path
                    d="M0 0h11v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar renders correctly without action button 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
        data-testid="close-button"
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 11 22"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902"
                />
              </g>
              <defs>
                <clippath
                  id="bc20b7afc9245ecd8673f572bfeb5871__a"
                >
                  <path
                    d="M0 0h11v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar renders correctly without close button 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            Action
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Snackbar renders with default props 1`] = `
<div>
  <div
    class="snackbar--container"
  >
    <div
      class="snackbar slide-up slide-down"
    >
      <div
        class="snackbar__message--container"
      >
        <p
          class="quill-typography__body-text__size--sm__weight--regular__decoration--default quill-typography__color--default snackbar__message"
        >
          test message
        </p>
      </div>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            Action
          </p>
        </span>
      </button>
      <button
        class="quill-button quill-button__size--md quill__color--tertiary-white"
        data-state=""
        data-testid="close-button"
      >
        <span
          class="button-label"
        >
          <p
            class="quill-typography__body-text__size--sm__weight--bold__decoration--default white"
          >
            <svg
              height="22"
              role="img"
              viewBox="0 0 11 22"
              width="11"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m9.684 7.496-3.282 3.281 3.254 3.254a.6.6 0 0 1 0 .903.6.6 0 0 1-.902 0L5.473 11.68l-3.254 3.254a.6.6 0 0 1-.903 0c-.273-.246-.273-.657 0-.93L4.57 10.75 1.316 7.496c-.273-.246-.273-.656 0-.93.246-.246.657-.246.93 0L5.5 9.848l3.254-3.254c.246-.274.656-.274.93 0a.644.644 0 0 1 0 .902"
                />
              </g>
              <defs>
                <clippath
                  id="bc20b7afc9245ecd8673f572bfeb5871__a"
                >
                  <path
                    d="M0 0h11v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </p>
        </span>
      </button>
    </div>
  </div>
</div>
`;
